### DISCUSSION TITLE
Go True email not coming through

### DISCUSSION THREAD

**Discussion by zigomir** [supabase/supabase#1843] [Questions] [answered]
Similar to https://github.com/supabase/supabase/discussions/1236 but using a local setup with `docker-compose`.

In my application code I have

```js
const { error, data } = await supabase.auth.signIn({
  email: 'my-email@gmail.com',
})
```

When I execute it, I get `null` back and no email (I checked spam folder).
I'm using a generated `docker-compose.yml`, which for `auth` service renders these values for `SMTP` related environment variables:

```yaml
GOTRUE_SMTP_HOST: mail
GOTRUE_SMTP_PORT: 2500
GOTRUE_SMTP_USER: GOTRUE_SMTP_USER
GOTRUE_SMTP_PASS: GOTRUE_SMTP_PASS
GOTRUE_SMTP_ADMIN_EMAIL: admin@email.com
```

should this suppose to "just work™"? If not, a message would be very helpful.

**Comment by soedirgo** [comment, marked as answer]
Hey @zigomir, the SMTP server in the local setup is for testing only, so you can view the emails that would've been sent (it's on localhost:9000 by default), but it's not actually sent. You'll have to set up your own SMTP server to actually send the emails. We have some more work to do to make it easier, so that ideally you don't have to figure out what each of the env vars mean (for the auth server it's all documented [here](https://github.com/supabase/gotrue)).

About the `null` in the code, I believe you need to specify the password as well, see [here](https://supabase.io/docs/reference/javascript/auth-signin).

**Reply by zigomir** [reply]
Thanks @soedirgo! If I only read through all docker compose service names 🤦🏼‍♂️
Still, it would've been nice to see a bullet on that in https://supabase.io/docs/guides/local-development – might open a PR :)

**Reply by NixBiks** [reply]
Hi @soedirgo 

Is it possible to configure the SMTP in the local dev setup using the supabase CLI? Or isn't possible to use email login locally?

**Reply by soedirgo** [reply]
@mr-bjerre right now the auth server is set up to autoconfirm email confirmations. But we'll add either an [email testing server](https://github.com/supabase/cli/issues/71) or custom SMTP in the future.

**Comment by bhvngt** [comment]
@soedirgo I was trying to do similar thing by configuring custom smtp with docker-compose using .env file.

My configs are as follows

```.env
# Secrets

POSTGRES_PASSWORD=postgres
JWT_SECRET=your-super-secret-jwt-token-with-at-least-32-characters-long

# Auth

## General
SITE_URL=http://localhost:8000
ADDITIONAL_REDIRECT_URLS=
JWT_EXPIRY=3600
DISABLE_SIGNUP=false

## Email auth
ENABLE_EMAIL_SIGNUP=true
ENABLE_EMAIL_AUTOCONFIRM=false
SMTP_ADMIN_EMAIL=<admin email>
SMTP_HOST=<smtp server>
SMTP_PORT=587
SMTP_USER=<admin email>
SMTP_PASS=<password>
SMTP_SENDER_NAME=<admin email>

## DB port
POSTGRES_PORT=5432

```

```Auth.ts
const client: SupabaseClient = createClient(
	process.env?.SUPABASE_URL ?? "",
	process.env?.SUPABASE_SERVICE_ROLE_KEY ?? ""
);

export default class AuthCommand {
	public async signIn(): Promise<UserCredentials> {
		return await client.auth.signIn({ email: <user_email_id> });
	}
}
```

Would this work? or is this feature still [WIP](https://github.com/supabase/cli/issues/71)

**Reply by bhvngt** [reply]
Right now I am getting `Error sending magic link` error with server status `500` error.

**Reply by bhvngt** [reply]
There was an issue with my custom smtp configurations. After fixing that, I was able to make this work from my localhost using docker-compose.
